### ISSUE TITLE
Missing argument validation in slice

### ISSUE THREAD

**Issue by robinsierra** [vyperlang/vyper#1467]
### Version Information

* vyper version: 0.1.0b9

### What's your issue about?

The `slice` function seems to only partially validate its arguments:
```
@public
def slc(a: string[4], s: int128, l: int128) -> string[4]:
    return slice(a, start=s, len=l)
```
For arguments ("abcd", 3, 5) it correctly reverts, however with arguments ("abcd", -1, 3) it works and returns all zero bytes, while arguments ("abcd", 1, -1) lead to an out-of-gas exception.

### How can it be fixed?

Validate both arguments to `slice`.

**Comment by fubuloubu** [comment]
Still an issue

**Comment by iamdefinitelyahuman** [comment]
Fixed in #2033
